src/adapter/src/coord/sequencer.rs

```diff
@@ -588,13 +589,29 @@ impl<S: Append + 'static> Coordinator<S> {
         let connection_gid = self.catalog.allocate_user_id().await?;
         let mut connection = plan.connection.connection.clone();
 
-        if let mz_storage::types::connections::Connection::Ssh(ref mut ssh) = connection {
-            let keyset = SshKeyset::new()?;
-            self.secrets_controller
-                .ensure(connection_gid, &keyset.to_bytes())
-                .await?;
-
-            ssh.public_keys = Some(keyset.public_keys());
+        match connection {
+            mz_storage::types::connections::Connection::Ssh(ref mut ssh) => {
+                let keyset = SshKeyset::new()?;
+                self.secrets_controller
+                    .ensure(connection_gid, &keyset.to_bytes())
+                    .await?;
+
+                ssh.public_keys = Some(keyset.public_keys());
+            }
+            mz_storage::types::connections::Connection::AwsPrivateLink(ref privatelink) => {
+                self.cloud_resource_controller
+                    .as_ref()
+                    .expect("PrivateLink connections are only allowed in cloud.")
```

Comment:
If someone tries to create one of these connections locally will we panic or is an error returned somewhere earlier? I don't think we should panic and instead return some form of error with this message.

Reply:
I'll switch to `AdapterError::Unsupported` here, but at startup we aren't allowed to throw any errors at that point. It's probably ok to `unwrap` there, though, since they shouldn't have any PrivateLink connections at startup since we prevent that here.